hail/python/hailtop/hailctl/dataproc/submit.py

```diff
@@ -2,81 +2,67 @@
+from typing import Optional, List
 
+from . import gcloud
 
-async def main(args, pass_through_args):  # pylint: disable=unused-argument
-    print("Submitting to cluster '{}'...".format(args.name))
 
-    # create files argument
-    files = ''
-    if args.files:
-        files = args.files
+def submit(
+    name: str,
+    script: str,
+    files: str,
+    pyfiles: str,
+    properties: Optional[str],
+    gcloud_configuration: Optional[str],
+    dry_run: bool,
+    region: Optional[str],
+    pass_through_args: List[str],
+):
+    print("Submitting to cluster '{}'...".format(name))
 
     def _filter_pyfile(fname: str) -> bool:
         return not fname.endswith('.pyc')
 
     # If you only provide one (comma-sep) argument, and it's a zip file, use that file directly
-    if args.pyfiles and args.pyfiles.endswith('.zip') and ',' not in args.pyfiles:
-        # Adding the zip archive directly
-        pyfiles = args.pyfiles
-    else:
-        pyfiles = []
-        if args.pyfiles:
-            pyfiles.extend(args.pyfiles.split(','))
-        pyfiles.extend(os.environ.get('HAIL_SCRIPTS', '').split(':'))
+    if not (pyfiles and pyfiles.endswith('.zip') and ',' not in pyfiles):
+        pyfiles_list = []
         if pyfiles:
+            pyfiles_list.extend(pyfiles.split(','))
+        pyfiles_list.extend(os.environ.get('HAIL_SCRIPTS', '').split(':'))
+        if pyfiles_list:
             tfile = tempfile.mkstemp(suffix='.zip', prefix='pyscripts_')[1]
-            zipf = zipfile.ZipFile(tfile, 'w', zipfile.ZIP_DEFLATED)
-            for hail_script_entry in pyfiles:
-                if os.path.isfile(hail_script_entry) and _filter_pyfile(hail_script_entry):
-                    zipf.write(hail_script_entry, arcname=os.path.basename(hail_script_entry))
-                else:
-                    for root, _, pyfiles_walk in os.walk(hail_script_entry):
-                        for pyfile in pyfiles_walk:
-                            path = os.path.join(root, pyfile)
-                            if os.path.isfile(path) and _filter_pyfile(path):
-                                zipf.write(path,
-                                           os.path.relpath(path,
-                                                           os.path.join(hail_script_entry, '..')))
-            zipf.close()
+            with zipfile.ZipFile(tfile, 'w', zipfile.ZIP_DEFLATED) as zipf:
+                for hail_script_entry in pyfiles_list:
+                    if os.path.isfile(hail_script_entry) and _filter_pyfile(hail_script_entry):
+                        zipf.write(hail_script_entry, arcname=os.path.basename(hail_script_entry))
+                    else:
+                        for root, _, pyfiles_walk in os.walk(hail_script_entry):
+                            for pyfile in pyfiles_walk:
+                                path = os.path.join(root, pyfile)
+                                if os.path.isfile(path) and _filter_pyfile(path):
```

Comment:
Our previous used the `path` variable in `zip.write`'s arguments too, but I agree it's better this way — keep those lines the same as upstream.

Reply:
I manually rewrote this block to by a hybrid of above, part of the painful merge :sweat:
Doesn't look like we can feasibly revert back to upstream

Reply:
Yep, what I'm saying is I'm happy with it :+1: